tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -453,6 +451,221 @@ def validate_executor_placements(cls, executor_placements):
+
+  @classmethod
+  def validate_executor_placements(cls, executor_placements):
+    singleton_placements = [
+        placement_literals.SERVER,
+        placement_literals.AGGREGATOR,
+    ]
+    expected_placements = singleton_placements + [placement_literals.CLIENTS]
+    py_typecheck.check_type(executor_placements, dict)
+    for k, v in executor_placements.items():
+      if k is not None:
+        py_typecheck.check_type(k, placement_literals.PlacementLiteral)
+      py_typecheck.check_type(v, (list, executor_base.Executor))
+      if isinstance(v, list):
+        for e in v:
+          py_typecheck.check_type(e, executor_base.Executor)
+      for pl in expected_placements:
+        if pl not in executor_placements:
+          raise ValueError(
+              'Improper executor placements, expected executor stack for '
+              '{} placement.'.format(pl.name))
+      for pl in [None] + singleton_placements:
+        if pl in executor_placements:
+          ex = executor_placements[pl]
+          if isinstance(ex, list):
+            pl_cardinality = len(ex)
+            if pl_cardinality != 1:
+              raise ValueError(
+                  'Unsupported cardinality for placement "{}": {}.'.format(
+                      pl, pl_cardinality))
+
+  async def federated_value_at_server(self, arg):
+    return await self._place(arg, placement_literals.SERVER)
+
+  async def federated_value_at_clients(self, arg):
+    return await self._place(arg, placement_literals.CLIENTS)
+
+  async def federated_eval_at_server(self, arg):
+    return await self._eval(arg, placement_literals.SERVER, True)
+
+  async def federated_eval_at_clients(self, arg):
+    return await self._eval(arg, placement_literals.CLIENTS, False)
+
+  async def federated_apply(self, arg):
+    return await self._map(arg)
+
+  async def federated_map(self, arg):
+    return await self._map(arg, all_equal=False)
+
+  async def federated_map_all_equal(self, arg):
+    return await self._map(arg, all_equal=True)
+
+  async def federated_broadcast(self, arg):
+    py_typecheck.check_type(arg.type_signature, computation_types.FederatedType)
+    py_typecheck.check_type(arg.internal_representation, list)
+    if not arg.type_signature.all_equal:
+      raise ValueError('Cannot broadcast a non all_equal value.')
+    if len(arg.internal_representation) != 1:
+      raise ValueError(
+          'Cannot broadcast a with a non-singleton representation.')
+    val = await arg.internal_representation[0].compute()
+    return FederatingExecutorValue(
+        await asyncio.gather(*[
+            c.create_value(val, arg.type_signature.member)
+            for c in self._get_child_executors(placement_literals.CLIENTS)
+        ]), type_factory.at_clients(arg.type_signature.member, all_equal=True))
+
+  async def federated_zip_at_server(self, arg):
+    return await self._zip(arg, placement_literals.SERVER, all_equal=True)
+
+  async def federated_zip_at_clients(self, arg):
```

Comment:
they're going all in with unique intrinsics, huh?

Reply:
Yes indeed